### PR TITLE
DIV-4303 - added new desertion fields (time together/lived apart since)

### PR DESCRIPTION
**Pull request by hosainnet**
# Description

To be able to used for desertion-specific questions and show them separately in CCD

(please check collapsed files)

Fixes #DIV-4303

## Type of change

- [x] New feature (non-breaking change which adds functionality)

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/divorce/caseformatterservice/strategy/reasonfordivorce/ReasonForDivorceStrategy.java [hunk 1]
```diff
@@ -1,10 +1,15 @@
...
 
+import uk.gov.hmcts.reform.divorce.caseformatterservice.domain.model.ccd.CoreCaseData;
 import uk.gov.hmcts.reform.divorce.caseformatterservice.domain.model.usersession.DivorceSession;
 
 public interface ReasonForDivorceStrategy {
 
     boolean accepts(String reasonForDivorce);
 
     String deriveStatementOfCase(DivorceSession divorceSession);
+
```
Comment (gurpiarbassi) [inline src/main/java/uk/gov/hmcts/reform/divorce/caseformatterservice/strategy/reasonfordivorce/ReasonForDivorceStrategy.java]: could you look at the PMD warnings generated? They're minor things.

File: src/main/java/uk/gov/hmcts/reform/divorce/caseformatterservice/strategy/reasonfordivorce/ReasonForDivorceContext.java [hunk 2]
```diff
@@ -23,4 +26,16 @@ public String deriveStatementOfWork(DivorceSession divorceSession) {
             .map(s -> s.deriveStatementOfCase(divorceSession))
             .collect(Collectors.joining());
     }
+
+    public void setLivedApartFieldsFromDivorceSession(DivorceSession divorceSession, CoreCaseData coreCaseData) {
```
Comment (qzhou-hmcts) [inline src/main/java/uk/gov/hmcts/reform/divorce/caseformatterservice/strategy/reasonfordivorce/ReasonForDivorceContext.java]: Unit test missing

**Comment by qzhou-hmcts** [comment]
Can you add Unit test for all the new classes?

File: src/main/java/uk/gov/hmcts/reform/divorce/caseformatterservice/strategy/reasonfordivorce/ReasonForDivorceContext.java (hunk 2 above)
Comment (hosainnet) [reply src/main/java/uk/gov/hmcts/reform/divorce/caseformatterservice/strategy/reasonfordivorce/ReasonForDivorceContext.java]: If you look at the coverage report, we have more or less 100% coverage https://codecov.io/gh/hmcts/div-case-data-formatter/pull/102/diff?src=pr&el=tree-more

Happy to add more tests but it'll be effectively duplicate

File: src/main/java/uk/gov/hmcts/reform/divorce/caseformatterservice/strategy/reasonfordivorce/ReasonForDivorceStrategy.java (hunk 1 above)
Comment (hosainnet) [reply src/main/java/uk/gov/hmcts/reform/divorce/caseformatterservice/strategy/reasonfordivorce/ReasonForDivorceStrategy.java]: Which ones, the deprecated warnings due to tool upgrade?

Comment (gurpiarbassi) [reply src/main/java/uk/gov/hmcts/reform/divorce/caseformatterservice/strategy/reasonfordivorce/ReasonForDivorceStrategy.java]: no if you run PMD locally it produces a report of violations in build/reports/pmd

Comment (hosainnet) [reply src/main/java/uk/gov/hmcts/reform/divorce/caseformatterservice/strategy/reasonfordivorce/ReasonForDivorceStrategy.java]: Done, thanks. I was only running checkstyleMain but the warnings were for tests.
On a semi-related note, if we implement the fixes for rules in https://github.com/hmcts/div-case-maintenance-service/pull/87/commits/90834911167dfa4b6f5875486cc5fde031a04398 we get 800+ new warnings, which we probably need to address separately

Comment (gurpiarbassi) [reply src/main/java/uk/gov/hmcts/reform/divorce/caseformatterservice/strategy/reasonfordivorce/ReasonForDivorceStrategy.java]: Yeh I believe Tony has done that on some other repo. So could pinch that at some point.

Comment (gurpiarbassi) [reply src/main/java/uk/gov/hmcts/reform/divorce/caseformatterservice/strategy/reasonfordivorce/ReasonForDivorceStrategy.java]: I think you had some warnings in pmdMain too. About missing java doc on empty default methods.

File: src/main/java/uk/gov/hmcts/reform/divorce/caseformatterservice/strategy/reasonfordivorce/ReasonForDivorceContext.java (hunk 2 above)
Comment (hosainnet) [reply src/main/java/uk/gov/hmcts/reform/divorce/caseformatterservice/strategy/reasonfordivorce/ReasonForDivorceContext.java]: Added 👍

File: src/main/java/uk/gov/hmcts/reform/divorce/caseformatterservice/strategy/reasonfordivorce/ReasonForDivorceStrategy.java (hunk 1 above)
Comment (hosainnet) [reply src/main/java/uk/gov/hmcts/reform/divorce/caseformatterservice/strategy/reasonfordivorce/ReasonForDivorceStrategy.java]: Done 👍

**Comment by nsingamsetty** [comment]
PR ready to Merge.
